src/main/java/hudson/plugins/jobConfigHistory/JobConfigHistory.java

```diff
@@ -118,28 +123,22 @@ public class JobConfigHistory extends Plugin {
      */
     private boolean showChangeReasonCommentWindow = true;
 
-    @Override
-    public void start() throws Exception {
+    @DataBoundConstructor
+    @Restricted(NoExternalUse.class)
+    public JobConfigHistory() {
         load();
-        loadRegexpPatterns();

```

Comment:
Are you sure that this can be removed?
Isn't it required when the XML config file is loaded from disk without any JCasC involved?

Reply:
Yeah you are right. I think it is still required.

So actually there is an existing test in `JobConfigHistoryTest` that is confusing, asserting that the field is `null` https://github.com/jenkinsci/job-config-history-plugin/blob/1171.v04b_66d78555e/src/test/java/hudson/plugins/jobConfigHistory/JobConfigHistoryTest.java#L295. Will probably need to adjust this one.

Reply:
Had to fix 2 tests that included system config that should be excluded by the pattern:

* https://github.com/jenkinsci/job-config-history-plugin/blob/1171.v04b_66d78555e/src/test/java/hudson/plugins/jobConfigHistory/FileHistoryDaoTest.java#L388
* https://github.com/jenkinsci/job-config-history-plugin/blob/1171.v04b_66d78555e/src/test/java/hudson/plugins/jobConfigHistory/FileHistoryDaoTest.java#L444

Reply:
For me the previous check for `null` in `JobConfigHistoryTest` is not confusing. It based on the fact that the `excludeRegexpPattern` field was always `null` until the configuration was saved the first time via the web interface or the configuration file contained a value.

You have changed the behaviour that now this field always contains the default exclude pattern until overwritten. I think this is the better implementation.